docs/docs/integrations/tools/desearch.ipynb

```diff
@@ -0,0 +1,319 @@
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "AI-Powered Search Engine and API for Advanced Data Discovery [Desearch](https://desearch.ai).\n",
+    "\n",
+    "Welcome to the comprehensive guide on integrating Desearch with LangChain to enable advanced Retrieval-Augmented Generation (RAG) and agent-style workflows. Desearch is a cutting-edge, privacy-first search API that excels in delivering relevant, real-time web and Twitter content, specifically optimized for use in Large Language Model (LLM) applications.\n",
+    "\n",
+    "First, get an Desearch API key and add it as an environment variable. Get $5 free credit (plus more by completing certain actions like making your first search) by [signing up here](https://console.desearch.ai)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    ".env DESEARCH_API_KEY=your_api_key_here"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from dotenv import load_dotenv\n",
+    "\n",
+    "load_dotenv()\n",
+    "# Set your API key as an environment variable"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "And install the integration package"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pip install -U langchain-desearch"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Instantiation"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 🛠️ Tools Overview\n",
+    "\n",
+    "Desearch provides a suite of modular tools compatible with LangChain, each designed for specific use cases:"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 🔎 Search Tools\n",
+    "\n",
+    "* **DesearchTool**: A comprehensive tool for searching across web, AI, and Twitter posts.\n",
+    "* **BasicWebSearchTool**: A lightweight tool focused solely on web searches.\n",
+    "* **BasicTwitterSearchTool**: A specialized tool for searching tweets with advanced filtering options."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Invocation"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 🧪 Quick Examples"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Basic Usage"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here's a simple example to demonstrate how to use the DesearchTool for web searches"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain_desearch.tools import DesearchTool\n",
```

Comment:
Please only include documentation about the tool. You can use `tool.invoke()` to show how to invoke the tool.

Remove the documentation about the agent, including the create_agent code from the package as it's using deprecated agent functionality and it's not something we'd like to recommend to users.